packages/microservices/client/client-kafka.ts

```diff
@@ -189,17 +189,18 @@ export class ClientKafka extends ClientProxy {
     return this.consumerAssignments;
   }
 
-  protected dispatchEvent(packet: OutgoingEvent): Promise<any> {
+  protected async dispatchEvent(packet: OutgoingEvent): Promise<any> {
     const pattern = this.normalizePattern(packet.pattern);
-    const outgoingEvent = this.serializer.serialize(packet.data);
+    const outgoingEvent = await this.serializer.serialize(packet.data);
     const message = Object.assign(
       {
         topic: pattern,
         messages: [outgoingEvent],
       },
       this.options.send || {},
     );
-    return this.producer.send(message);
+
+    return await this.producer.send(message);
```

Comment:
```suggestion
    return this.producer.send(message);
```
You don't need `await` here